### PR TITLE
[common]: Housekeeping

### PR DESCRIPTION
**Pull request by janfuhrer**
Signed-off-by: Jan Fuhrer <janfuhrer@mailbox.org>

**What this PR does**:

- improve `values.schema.json` & ci test-values
- disable networkpolicy by default

**Notes for Reviewer**:

**Checklist**:

- [x] Pull Request title in format `[chart]: Changed Something`
- [x] Updated documentation in the  `README.md.gotmpl` file and executed `helm-docs` 
- [x] Chart Version bumped
- [x] All commits are signed-off

### CODE REVIEW

File: charts/common/values.schema.json [hunk 1]
```diff
@@ -1,6 +1,5 @@
 {
   "$schema": "http://json-schema.org/draft-07/schema",
-  "$id": "https://s3.mgmtbi.ch/json-schemas/helm/common/x.y.z/values.schema.json",
```
Comment (janfuhrer) [inline charts/common/values.schema.json:3]: I'm not sure if this reference is needed, linting works locally without this line

**Comment by janfuhrer** [comment]
Do we need to publish the `values.schema.json` with the helm-chart release or is it sufficient to reference in a derived chart like this:

```json
{
    "$schema": "http://json-schema.org/draft-07/schema",
    "type": "object",
    "$ref": "https://raw.githubusercontent.com/bedag/helm-charts/common-7.0.1/charts/common/values.schema.json"
}

```

File: charts/common/values.schema.json (hunk 1 above)
Comment (adberger) [reply charts/common/values.schema.json:3]: This is probably not needed currently, see use cases:
https://json-schema.org/understanding-json-schema/structuring.html

**Comment by adberger** [comment]
> Do we need to publish the `values.schema.json` with the helm-chart release or is it sufficient to reference in a derived chart like this:
> 
> ```
> {
>     "$schema": "http://json-schema.org/draft-07/schema",
>     "type": "object",
>     "$ref": "https://raw.githubusercontent.com/bedag/helm-charts/common-7.0.1/charts/common/values.schema.json"
> }
> ```

No need to publish it, since we can get the `values.schema.json` by version with the help of the tag references.

**Comment by adberger** [comment]
@janfuhrer 
Could you also fix the name of the following ids:

- https://github.com/janfuhrer/helm-charts/blob/master/charts/common/values.schema.json#L217
- https://github.com/janfuhrer/helm-charts/blob/master/charts/common/values.schema.json#L913

**Comment by janfuhrer** [comment]
> @janfuhrer Could you also fix the name of the following ids:
> 
> * https://github.com/janfuhrer/helm-charts/blob/master/charts/common/values.schema.json#L217
> * https://github.com/janfuhrer/helm-charts/blob/master/charts/common/values.schema.json#L913

done, I am still not sure if we need this `$id` field at all or if we can leave it out completely

**Comment by adberger** [comment]
> > @janfuhrer Could you also fix the name of the following ids:
> > 
> > * https://github.com/janfuhrer/helm-charts/blob/master/charts/common/values.schema.json#L217
> > * https://github.com/janfuhrer/helm-charts/blob/master/charts/common/values.schema.json#L913
> 
> done, I'm still not sure if we need this `$id` fields at all or if we can leave it out completely

I'm not sure either, just test it locally.
Build a helm chart with common as dependency.
Set $ref in `values.schema.json` in new helm chart to _https://raw.githubusercontent.com/janfuhrer/helm-charts/master/charts/common/values.schema.json_

**Comment by janfuhrer** [comment]
> > > @janfuhrer Could you also fix the name of the following ids:
> > > 
> > > * https://github.com/janfuhrer/helm-charts/blob/master/charts/common/values.schema.json#L217
> > > * https://github.com/janfuhrer/helm-charts/blob/master/charts/common/values.schema.json#L913
> > 
> > 
> > done, I'm still not sure if we need this `$id` fields at all or if we can leave it out completely
> 
> I'm not sure either, just test it locally. Build a helm chart with common as dependency. Set $ref in `values.schema.json` in new helm chart to _https://raw.githubusercontent.com/janfuhrer/helm-charts/master/charts/common/values.schema.json_

I've removed all `$id` fields and tested as follows:

- uncomment all includes in `includes.yaml`
- `helm template . -f ci/values.test.yaml` works as expected
- add something wrong:
```yaml
components:
  foo: bar
```
--> fails as assumed:
```yaml
Error: values don't meet the specifications of the schema(s) in the following chart(s):
common:
- components.foo: Invalid type. Expected: object, given: string
```

**Review by adberger — APPROVED** [review]
lgtm

**Pull request merged**